File: libs/sdk-python/src/daytona/common/disk.py
# Copyright 2025 Daytona Platforms Inc.
# SPDX-License-Identifier: Apache-2.0

from daytona_api_client import DiskDto


class Disk(DiskDto):
    """Represents a Daytona Disk which is persistent storage for Sandboxes.

    Attributes:
        id (StrictStr): Unique identifier for the Disk.
        name (StrictStr): Name of the Disk.
        organization_id (StrictStr): Organization ID of the Disk.
        size (int): Disk size in GB.
        state (DiskState): State of the Disk.
        runner_id (Optional[StrictStr]): Runner ID where disk is located.
        error_reason (Optional[StrictStr]): Error reason if in error state.
        created_at (StrictStr): Date and time when the Disk was created.
        updated_at (StrictStr): Date and time when the Disk was last updated.
    """

    @classmethod
    def from_dto(cls, dto: DiskDto) -> "Disk":
        return cls(**dto.__dict__)

File: apps/runner/pkg/api/dto/disk.go
// Copyright 2025 Daytona Platforms Inc.
// SPDX-License-Identifier: AGPL-3.0

package dto

type DiskDTO struct {
	DiskId    string `json:"diskId"`
	MountPath string `json:"mountPath"`
} //	@name	DiskDTO

type DiskInfoDTO struct {
	Name         string `json:"name"`
	SizeGB       int64  `json:"sizeGB"`
	ActualSizeGB int64  `json:"actualSizeGB"`
	Created      string `json:"created"`
	Modified     string `json:"modified"`
	IsMounted    bool   `json:"isMounted"`
	MountPath    string `json:"mountPath"`
	InS3         bool   `json:"inS3"`
	Checksum     string `json:"checksum"`
} //	@name	DiskInfoDTO
